policy/aggregation_type_options.go

```diff
@@ -283,6 +289,29 @@ func (o *options) initPools() {
 	o.quantilesPool.Init()
 }
 
+func (o *options) Validate() error {
+	if err := o.validateUniqSuffixForMetricType(o.counterSuffixes, metric.CounterType); err != nil {
+		return err
+	}
+	if err := o.validateUniqSuffixForMetricType(o.timerSuffixes, metric.TimerType); err != nil {
+		return err
+	}
+	return o.validateUniqSuffixForMetricType(o.gaugeSuffixes, metric.GaugeType)
+}
+
+func (o *options) validateUniqSuffixForMetricType(suffixes [][]byte, t metric.Type) error {
+	m := make(map[string]int, len(suffixes))
+	for aggType, suffix := range suffixes {
+		s := string(suffix)
+		if existAggType, ok := m[s]; ok {
+			return fmt.Errorf("invalid options, found duplicated suffix: '%s' for aggregation type %v and %v for metric type: %s",
+				s, AggregationType(aggType), AggregationType(existAggType), t.String())
+		}
+		m[string(suffix)] = aggType
```

Comment:
nit: Use `m[s] = aggType` instead to save an allocation.

Reply:
sg